flutter_todo/lib/components/todo_list.dart

```diff
@@ -14,68 +14,74 @@ class TodoList extends StatefulWidget {
-          for (final deletionIndex in data.deleted) {
-            final toDie = _itemViewModels
-                .removeAt(deletionIndex); // update view model collection
-            _myListKey.currentState?.removeItem(deletionIndex,
-                (context, animation) {
-              return ItemCard(toDie, animation);
-            });
-          }
+  @override
+  void dispose() {
+    super.dispose();
+  }
 
-          // Handle inserts
-          for (final insertionIndex in data.inserted) {
-            _itemViewModels.insert(
-                insertionIndex, ItemViewModel(realm, items[insertionIndex]));
-            _myListKey.currentState?.insertItem(insertionIndex);
-          }
+  @override
+  Widget build(BuildContext context) {
+    final realmServices = Provider.of<RealmServices>(context);
+    return Stack(
+      children: [
+        Column(
+            children: [
+            styledBox(
+              child: Row(
+                  children: [
+                    const Expanded(child: Text("Show All Tasks", textAlign: TextAlign.right)),
+                    Switch(
+                      value: realmServices.showAll,
+                      onChanged: (value) async => await realmServices.switchSubscription(value),
+                    ),
+                  ],
+              ),
+            ),
+            Expanded(
+              child: Padding(
+                padding: const EdgeInsets.fromLTRB(16, 0, 16, 0),
+                child: StreamBuilder<RealmResultsChanges<Item>>(
+                  stream: realmServices.realm.all<Item>().changes,
+                  builder: (context, snapshot) {
+                    final data = snapshot.data;
 
-          // Handle modifications
-          for (final modifiedIndex in data.modified) {
-            _itemViewModels[modifiedIndex] =
-                ItemViewModel(realm, items[modifiedIndex]);
-          }
+                    if (data == null) return waitingIndicator();
 
-          // Handle initialization (or any mismatch really, but that shouldn't happen)
-          if (items.length != _itemViewModels.length) {
-            _itemViewModels.insertAll(
-                0, items.map((item) => ItemViewModel(realm, item)));
-            _itemViewModels.length = items.length;
-          }
+                    final results = data.results;
+                    return ListView.builder(
+                      shrinkWrap: true,
+                      itemCount: results.realm.isClosed ? 0 : results.length,
+                      itemBuilder: (context, index) => results[index].isValid ? TodoItem(results[index]) : Container(),
+                    );
+                  },
+                ),
+              ),
+            ),
+            styledBox(
+              child: const Expanded(
+                child: Text(
+                  '''Log in with the same account on another device
+or simulator to see your list sync in realm-time''',
```

Comment:
also wonder if worth mentioning that can see it in Atlas as well

Reply:
The text appear at the bottom of the list. I think it will become too long.

Reply:
Or if you have an idea how to make it shorter.

Reply:
Fixed in https://github.com/mongodb-university/realm-template-apps/pull/110/commits/bde8d827ad01dbd8ff0ebd45a682a4048410c2b9